variantlib/protocols.py

```diff
@@ -84,3 +84,35 @@ def get_build_setup(
     ) -> dict[str, list[str]]:
         """Get build variables for a variant made of specified properties"""
         return {}
+
+
+@runtime_checkable
+class PluginDynamicType(Protocol):
+    """A protocol for plugin classes"""
+
+    @property
+    @abstractmethod
+    def namespace(self) -> VariantNamespace:
+        """Plugin namespace"""
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_all_features(self) -> list[VariantFeatureName]:
+        """Get all feature names for the plugin"""
+        raise NotImplementedError
+
+    @abstractmethod
+    def filter_and_sort_properties(
```

Comment:
Just a random thought but I do wonder if it would be easier or harder to have plugins return sort keys (or `None` for unsupported variants) instead, and handle sorting and filtering on `variantlib` end. May or may not involve less duplication.